cmd/minikube/cmd/start.go

```diff
@@ -19,56 +19,83 @@ package cmd
 import (
 	"fmt"
 	"os"
-	"strconv"
-	"strings"
+	"path/filepath"
 
 	units "github.com/docker/go-units"
 	"github.com/docker/machine/libmachine"
 	"github.com/docker/machine/libmachine/host"
 	"github.com/golang/glog"
-	ocfg "github.com/openshift/origin/pkg/cmd/cli/config"
-	"github.com/spf13/cobra"
-	"github.com/spf13/viper"
-	"k8s.io/kubernetes/pkg/api"
-	cfg "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
-
 	"github.com/minishift/minishift/pkg/minikube/cluster"
 	"github.com/minishift/minishift/pkg/minikube/constants"
-	"github.com/minishift/minishift/pkg/minikube/kubeconfig"
+	"github.com/minishift/minishift/pkg/minishift/cache"
 	"github.com/minishift/minishift/pkg/util"
+	"github.com/minishift/minishift/pkg/version"
+	dockerhost "github.com/openshift/origin/pkg/bootstrap/docker/host"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
 )
 
 const (
+	commandName = "start"
+
+	// minishift
 	isoURL                = "iso-url"
 	memory                = "memory"
 	cpus                  = "cpus"
 	humanReadableDiskSize = "disk-size"
 	vmDriver              = "vm-driver"
 	openshiftVersion      = "openshift-version"
 	hostOnlyCIDR          = "host-only-cidr"
-	deployRegistry        = "deploy-registry"
-	deployRouter          = "deploy-router"
+
+	// cluster up
 )
 
 var (
 	dockerEnv        []string
 	insecureRegistry []string
 	registryMirror   []string
+	openShiftEnv     []string
 )
 
 // startCmd represents the start command
 var startCmd = &cobra.Command{
-	Use:   "start",
+	Use:   commandName,
 	Short: "Starts a local OpenShift cluster.",
 	Long: `Starts a local OpenShift cluster using Virtualbox. This command
```

Comment:
[Not part of review/blocker]
We have to change this description because virtualbox is not only provider which minishift uses.